content/en/blog/2022/opamp/index.md

```diff
@@ -0,0 +1,239 @@
+	fmt.Println("Starting server on port 8080")
+	log.Fatal(http.ListenAndServe(":8080", nil))
+}
+```
+
+Next, add a basic named “effective.yaml” configuration file. Place it in the
+same folder as our main.go file with this configuration:
+
+```yaml
+instrument: false
+```
+
+Let’s add a basic configuration handler to our server:
+
+```go
+package main
+
+import (
+   "fmt"
+   "gopkg.in/yaml.v3"
+   "io/ioutil"
+   "log"
+   "net/http"
+   "path/filepath"
+)
+
+type configurations struct {
+   Instrument bool
+}
+
+func httpHandler(w http.ResponseWriter, r *http.Request) {
+   fmt.Fprintf(w, "Hi! This action could create a trace!")
+}
+
+func main() {
+   filename, _ := filepath.Abs("./effective.yaml")
+   yamlFile, _ := ioutil.ReadFile(filename)
+
+   var config configurations
+   yaml.Unmarshal(yamlFile, &config)
+
+   handler := http.HandlerFunc(httpHandler)
+   http.Handle("/", handler)
+   fmt.Println("Starting server on port 8080")
+   log.Fatal(http.ListenAndServe(":8080", nil))
+}
+```
+
+Sorry, no error handling here. Next, let's wrap our handler with instrumentation
+and condition it with our configuration file. Something like this:
+
+```go
+package main
+
+import (
+   "context"
+   "fmt"
+   "go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp"
+   "go.opentelemetry.io/otel"
+   "go.opentelemetry.io/otel/exporters/stdout/stdouttrace"
+   sdktrace "go.opentelemetry.io/otel/sdk/trace"
+   "go.opentelemetry.io/otel/trace"
+   "gopkg.in/yaml.v3"
+   "io/ioutil"
+   "log"
+   "net/http"
+   "os"
+   "path/filepath"
+)
+
+type configurations struct {
+   Instrument bool
+}
+
+var tracer trace.Tracer
+
+func newConsoleExporter() (sdktrace.SpanExporter, error) {
+   return stdouttrace.New(
+      stdouttrace.WithWriter(os.Stdout),
+      stdouttrace.WithPrettyPrint(),
+   )
+}
+
+func httpHandler(w http.ResponseWriter, r *http.Request) {
+   fmt.Fprintf(w, "Hi! This action could create a trace!")
+}
+
+func setHandler(handler http.Handler, config configurations) http.Handler {
+   if config.Instrument {
+      return otelhttp.NewHandler(handler, "instrumentation activated by OpAMP")
+   }
+   return http.HandlerFunc(httpHandler)
+}
+
+func main() {
+   filename, _ := filepath.Abs("./effective.yaml")
+   yamlFile, _ := ioutil.ReadFile(filename)
+
+   var config configurations
+   yaml.Unmarshal(yamlFile, &config)
+
+   exp, _ := newConsoleExporter()
+   tp := sdktrace.NewTracerProvider(sdktrace.WithBatcher(exp))
+   defer func() { _ = tp.Shutdown(context.Background()) }()
+
+   otel.SetTracerProvider(tp)
+
+   tracer = tp.Tracer("ControlledOpAMPAgentDemo")
+   handler := http.HandlerFunc(httpHandler)
+   http.Handle("/", setHandler(handler, config))
+   fmt.Println("Starting server on port 8080")
+   log.Fatal(http.ListenAndServe(":8080", nil))
+}
+```
+
+Build it with `go build .` Now if we run it and open http://localhost:8080/ on a
+browser, nothing special will be shown. It’s time to add some opamp. Git clone
```

Comment:
shown. <newline><newline> It's time ...

Create a new paragraph

Reply:
Also suggest marking up the URL for readability, eg 'http://localhost:8080/'

Reply:
replaced by https://github.com/open-telemetry/opentelemetry.io/pull/1847/files#r994285865